src/model/impl/TaskManager.cpp

```diff
@@ -167,6 +167,28 @@ std::vector<TaskDTO> TaskManager::ShowChild(const TaskId& parent_id)
     }
     return result;
 }
+std::vector<TaskDTO> TaskManager::ShowTasksWithLabel(const std::string& label)
+{
+    auto result = std::vector<TaskDTO>{};
+
+    for (const auto& elem: this->tasks_)
+    {
+        auto task = elem.second.GetTask();
+        auto has_label = std::count_if(task.labels().cbegin(), task.labels().cend(),
+                                       [&label](const auto& tmp_label)
+                                       {
+                                           return tmp_label == label;
+                                       });
+        if (has_label)
+        {
+            auto tmp = CreateTaskDTO(elem.first, elem.second.GetTask());
```

Comment:
Why `ShowTasksWithLabel` returns tasks without their parent ids? Obviously, it was done in order to show them without hierarchy on UI, but, first of all it isn't obvious (how could I know that exactly this method returns tasks without their parents?), and why `Model` is responsible for selecting the way this tasks will be shown for user?